### ISSUE TITLE
--> Image not liked: b'Unavailable Page' Raspberry PI 4 Ubuntu 21.10

### ISSUE THREAD

**Issue by r4spberry13** [InstaPy/InstaPy#6416]
Hello,

I have some issues when i run quickstart.py on my raspberry pi4  running ubuntu 21.10

After lot of installation issues i finally run my quickstart.py file

I use Firfefox-esr 78.10 and arm version of geckodriver.

When i launch the script,  some errors appear :
`WARNING [2021-12-02 21:38:35] [myaccount]  - Hide Selenium Extension: error
`
`WARNING [2021-12-02 21:47:05] [myaccount]  Unavailable Page: b'https://www.instagram.com/p/CW9RrwFBK7L/'
INFO [2021-12-02 21:47:05] [myaccount]  --> Image not liked: b'Unavailable Page'
`

## InstaPy configuration
`# imports
from instapy import InstaPy
from instapy import smart_run

# login credentials
insta_username = 'account'
insta_password = 'pass'

comments = ['Super photo !',
'Top !',
'Jolie Cliché !',
'Nice pic !',
'Nice shot !',
'Nice!',
'So clean !',
'Propre !',
'sympa',
'jolie',
'interesting !']

# get an InstaPy session!
# set headless_browser=True to run InstaPy in the background
session = InstaPy(username=insta_username,
                  password=insta_password,
                  headless_browser=False)

with smart_run(session):
  """ Activity flow """		
  # general settings		
  session.set_dont_include(["friend1", "friend2", "friend3"])		
  
  # activity		
  session.like_by_tags(["z50r", "hondamonkey", "hondadax", "hondaz50r", "st70", "z50j", "crf50", "crf70", "xr50", "fast50s", "sano", "takegawa", "kitaco"], amount=7)
  session.set_do_follow(True, percentage=10)
  
  # Joining Engagement Pods
  session.set_do_comment(enabled=True, percentage=15)
  session.set_comments(comments)
  session.join_pods(topic='minibike', engagement_mode='no_comments')`

Thank  you for the time to read.

**Comment by CodeMasterBond** [comment]
Make sure you have the correct login and password info and also make sure two-factor authentication is disabled. The error you are receiving looks as though the bot is not able to login to the account. Therefore attempting to like and follow accounts useless.

**Comment by r4spberry13** [comment]
Pass and login are changed to publish on github but the bot launch correctly the window with instagram, connection is good i can see the bot checking pictures but just not liking.

I have check and 2fa auth is not enabled.

Thanks for you answer, i hope we will fix this issue :)

**Comment by CodeMasterBond** [comment]
Ah, might've found your issue. Add ``interact= True`` at the end of the ``like_by_tag`` line. So an example of this looks like...
``bot.like_by_tags(hashtags, amount=10, interact=True)``

**Comment by r4spberry13** [comment]
I've added the code but infortunally it doesn't work, the same error appear.

I've see an error that appear also before adding the Interact=True

 `/home/ubu/.local/lib/python3.9/site-packages/selenium/webdriver/remote/webelement.py:426: UserWarning: find_elements_by_* commands are deprecated. Please use find_elements() instead
  warnings.warn("find_elements_by_* commands are deprecated. Please use find_elements() instead")
`

**Comment by r4spberry13** [comment]
the results


`INFO [2021-12-02 23:57:02] [myaccount]  Like# [1/14]
INFO [2021-12-02 23:57:02] [myaccount]  https://www.instagram.com/p/CW-7d8MhFWs/
WARNING [2021-12-02 23:57:07] [myaccount]  Unavailable Page: b'https://www.instagram.com/p/CW-7d8MhFWs/'
INFO [2021-12-02 23:57:07] [myaccount]  --> Image not liked: b'Unavailable Page'
INFO [2021-12-02 23:57:07] [myaccount]  Like# [2/14]
INFO [2021-12-02 23:57:07] [myaccount]  https://www.instagram.com/p/CW-5H8_pNtd/
WARNING [2021-12-02 23:57:13] [myaccount]  Unavailable Page: b'https://www.instagram.com/p/CW-5H8_pNtd/'
INFO [2021-12-02 23:57:13] [myaccount]  --> Image not liked: b'Unavailable Page'
INFO [2021-12-02 23:57:13] [myaccount]  Like# [3/14]
INFO [2021-12-02 23:57:13] [myaccount]  https://www.instagram.com/p/CW_nNgMvqsX/
WARNING [2021-12-02 23:57:18] [myaccount]  Unavailable Page: b'https://www.instagram.com/p/CW_nNgMvqsX/'
INFO [2021-12-02 23:57:18] [myaccount]  --> Image not liked: b'Unavailable Page'
INFO [2021-12-02 23:57:18] [myaccount]  Like# [4/14]
INFO [2021-12-02 23:57:18] [myaccount]  https://www.instagram.com/p/CW_jLuOBUsJ/
WARNING [2021-12-02 23:57:24] [myaccount]  Unavailable Page: b'https://www.instagram.com/p/CW_jLuOBUsJ/'
INFO [2021-12-02 23:57:24] [myaccount]  --> Image not liked: b'Unavailable Page'
INFO [2021-12-02 23:57:24] [myaccount]  Like# [5/14]
INFO [2021-12-02 23:57:24] [myaccount]  https://www.instagram.com/p/CW_ruARhqUP/
WARNING [2021-12-02 23:57:30] [myaccount]  Unavailable Page: b'https://www.instagram.com/p/CW_ruARhqUP/'
INFO [2021-12-02 23:57:30] [myaccount]  --> Image not liked: b'Unavailable Page'
INFO [2021-12-02 23:57:30] [myaccount]  Like# [6/14]
INFO [2021-12-02 23:57:30] [myaccount]  https://www.instagram.com/p/CW_RUr5MG4b/
WARNING [2021-12-02 23:57:35] [myaccount]  Unavailable Page: b'https://www.instagram.com/p/CW_RUr5MG4b/'
INFO [2021-12-02 23:57:35] [myaccount]  --> Image not liked: b'Unavailable Page'
INFO [2021-12-02 23:57:35] [myaccount]  Like# [7/14]
INFO [2021-12-02 23:57:35] [myaccount]  https://www.instagram.com/p/CW-7LxOLdDn/
WARNING [2021-12-02 23:57:40] [myaccount]  Unavailable Page: b'https://www.instagram.com/p/CW-7LxOLdDn/'
INFO [2021-12-02 23:57:40] [myaccount]  --> Image not liked: b'Unavailable Page'
INFO [2021-12-02 23:57:40] [myaccount]  Like# [8/14]
INFO [2021-12-02 23:57:40] [myaccount]  https://www.instagram.com/p/CW_fe1Trl1l/
WARNING [2021-12-02 23:57:45] [myaccount]  Unavailable Page: b'https://www.instagram.com/p/CW_fe1Trl1l/'
INFO [2021-12-02 23:57:45] [myaccount]  --> Image not liked: b'Unavailable Page'
INFO [2021-12-02 23:57:45] [myaccount]  Like# [9/14]
INFO [2021-12-02 23:57:45] [myaccount]  https://www.instagram.com/p/CW_ej0JhJ9G/
WARNING [2021-12-02 23:57:51] [myaccount]  Unavailable Page: b'https://www.instagram.com/p/CW_ej0JhJ9G/'
INFO [2021-12-02 23:57:51] [myaccount]  --> Image not liked: b'Unavailable Page'
INFO [2021-12-02 23:57:51] [myaccount]  Like# [10/14]
INFO [2021-12-02 23:57:51] [myaccount]  https://www.instagram.com/p/CW-7T4XgQrQ/
WARNING [2021-12-02 23:57:57] [myaccount]  Unavailable Page: b'https://www.instagram.com/p/CW-7T4XgQrQ/'
INFO [2021-12-02 23:57:57] [myaccount]  --> Image not liked: b'Unavailable Page'
INFO [2021-12-02 23:57:57] [myaccount]  Like# [11/14]
INFO [2021-12-02 23:57:57] [myaccount]  https://www.instagram.com/p/CW_l4OPvPSs/
WARNING [2021-12-02 23:58:02] [myaccount]  Unavailable Page: b'https://www.instagram.com/p/CW_l4OPvPSs/'
INFO [2021-12-02 23:58:02] [myaccount]  --> Image not liked: b'Unavailable Page'
INFO [2021-12-02 23:58:02] [myaccount]  Like# [12/14]
INFO [2021-12-02 23:58:02] [myaccount]  https://www.instagram.com/p/CW_xY4vMs4O/
WARNING [2021-12-02 23:58:07] [myaccount]  Unavailable Page: b'https://www.instagram.com/p/CW_xY4vMs4O/'
INFO [2021-12-02 23:58:07] [myaccount]  --> Image not liked: b'Unavailable Page'
INFO [2021-12-02 23:58:07] [myaccount]  Like# [13/14]
INFO [2021-12-02 23:58:07] [myaccount]  https://www.instagram.com/p/CW_W9yvsNKu/
WARNING [2021-12-02 23:58:13] [myaccount]  Unavailable Page: b'https://www.instagram.com/p/CW_W9yvsNKu/'
INFO [2021-12-02 23:58:13] [myaccount]  --> Image not liked: b'Unavailable Page'
INFO [2021-12-02 23:58:13] [myaccount]  Like# [14/14]
INFO [2021-12-02 23:58:13] [myaccount]  https://www.instagram.com/p/CW_Mgejp92x/
WARNING [2021-12-02 23:58:18] [myaccount]  Unavailable Page: b'https://www.instagram.com/p/CW_Mgejp92x/'
INFO [2021-12-02 23:58:18] [myaccount]  --> Image not liked: b'Unavailable Page'
INFO [2021-12-02 23:58:18] [myaccount]  Tag: b'hondamonkey'
INFO [2021-12-02 23:58:18] [myaccount]  Liked: 0
INFO [2021-12-02 23:58:18] [myaccount]  Already Liked: 0
INFO [2021-12-02 23:58:18] [myaccount]  Commented: 0
INFO [2021-12-02 23:58:18] [myaccount]  Followed: 0
INFO [2021-12-02 23:58:18] [myaccount]  Inappropriate: 14
INFO [2021-12-02 23:58:18] [myaccount]  Not valid users: 0

ERROR [2021-12-02 23:58:18] [myaccount]  You have entered an invalid topic for pods, allowed topics are : ['general', 'fashion', 'food', 'travel', 'sports', 'entertainment']. Exiting...

INFO [2021-12-02 23:58:19] [myaccount]  Sessional Live Report:
	|> LIKED 0 images  |  ALREADY LIKED: 0
	|> COMMENTED on 0 images
	|> FOLLOWED 0 users  |  ALREADY FOLLOWED: 0
	|> UNFOLLOWED 0 users
	|> LIKED 0 comments
	|> REPLIED to 0 comments
	|> INAPPROPRIATE images: 14
	|> NOT VALID users: 0
	|> WATCHED 0 story(ies)  |  WATCHED 0 reel(s)

On session start was FOLLOWING 1109 users & had 6053 FOLLOWERS
[Session lasted 2.86 minutes]
OOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOO
INFO [2021-12-02 23:58:19] [myaccount]  Session ended!
oooooooooooooooooooooooooooooooooooooooooooooooooooooooooooo



ubu@ubu-desktop:~/bot/instapi$ ~
`

**Comment by CodeMasterBond** [comment]
> I've added the code but infortunally it doesn't work, the same error appear.
> 
> I've see an error that appear also before adding the Interact=True
> 
> `/home/ubu/.local/lib/python3.9/site-packages/selenium/webdriver/remote/webelement.py:426: UserWarning: find_elements_by_* commands are deprecated. Please use find_elements() instead warnings.warn("find_elements_by_* commands are deprecated. Please use find_elements() instead") `

The error you just showed is ok. It doesn't effect the program because the same shows up for me and works. I also just now realized that you are running instapy off the firefox enterprise version and a raspberry pi. I never ran instapy off my pi, (though I am going to try it over the weekend), so I figured that if I get the same error off my pi, then it's a pi problem I need to fix. What you can try to do is install the "regular" patch of firefox and run it. Once again, idk if it will fix the issue but it's worth a try. Lmk how it goes.

**Comment by r4spberry13** [comment]
after lot of try, it run with firefox 78.10 on my pi, what is the regular patch ?

Thank you a lot for your help, it's very complicated for me and i think this will help a lot of person who try to run on raspberry pi :)

**Comment by CodeMasterBond** [comment]
I'm glad its working and I agree, this issue will help the raspberry pi users greatly. Always happy to help!

Btw, when I say ``regular patch``, I mean this one... ``https://www.mozilla.org/en-US/firefox/new/``

**Comment by r4spberry13** [comment]
okay thank you i try this and i give you a return

**Comment by elulcao** [comment]
> I've added the code but infortunally it doesn't work, the same error appear.
> 
> I've see an error that appear also before adding the Interact=True
> 
> `/home/ubu/.local/lib/python3.9/site-packages/selenium/webdriver/remote/webelement.py:426: UserWarning: find_elements_by_* commands are deprecated. Please use find_elements() instead warnings.warn("find_elements_by_* commands are deprecated. Please use find_elements() instead")`

Working on it, check #6393

**Comment by CodeMasterBond** [comment]
@elulcao The error does not seem to make a difference with any of the commands I tested. Idk about others but for me, its simply just a message saying nothing.

**Comment by elulcao** [comment]
@CodeMasterBond 

`find_elements() warning message` is tracked in #6393
In my local I have the same problem for 'Unavailable Page', this is because I ran the bot several (I mean Several...) times in a short period of time. So I have to wait a while before executing again the bot.

This is the current UI for the failure, it means to me IG targeted my Acct 🤣:
![Captura de Pantalla 2021-12-02 a la(s) 10 30 44](https://user-images.githubusercontent.com/6373960/144541479-afdcee8d-892b-418f-a5bb-ac44e3f72ab3.png)

**Comment by CodeMasterBond** [comment]
Your explanation makes sense. How much followers did you gain till Instagram targeted your Acct? Lmao

**Comment by elulcao** [comment]
I cannot confirm new followers since my act is freeze, based on `followerNum.txt` under WorkSpace dir the total is a big 0:

```bash
2021-12-01 19:03 0
2021-12-02 10:35 0
2021-12-02 13:47 0
2021-12-02 17:48 0
2021-12-02 19:16 0
```

At this point IG targeted my IP (the IP from ISP), so I have to reboot the Router and wait some time to WA the block, after getting a new public IP the bot can be used again.

**Comment by CodeMasterBond** [comment]
Wow. That is really unfortunate. Have you ever heard of proxy chains? @elulcao

**Comment by elulcao** [comment]
> Wow. That is really unfortunate. Have you ever heard of proxy chains? @elulcao

yes, I have heard. The thing is that I prefer to create a new Acct instead of the proxy. I mean, I really like the process of growing up an acct, on the other hand, I can connect to the I can also connect to neighboring networks so as not to lose speed.

**Comment by CodeMasterBond** [comment]
Yes, that makes sense. Are you one of the co-founders of instapy btw?

**Comment by elulcao** [comment]
@CodeMasterBond 

😃 I'm a collaborator 👨‍💻 and `InstaPy` user 🤖 ❤️

**Comment by CodeMasterBond** [comment]
How do you become a collaborator? I am a active instapy user and am interesting in modifying and helping the project further.

**Comment by r4spberry13** [comment]
> > I've added the code but infortunally it doesn't work, the same error appear.
> > I've see an error that appear also before adding the Interact=True
> > `/home/ubu/.local/lib/python3.9/site-packages/selenium/webdriver/remote/webelement.py:426: UserWarning: find_elements_by_* commands are deprecated. Please use find_elements() instead warnings.warn("find_elements_by_* commands are deprecated. Please use find_elements() instead")`
> 
> Working on it, check #6393

i have tested to change requirements.txt 
`Update requirements.txt from selenium>=3.141.0 to selenium>=3.141.0,<= 3.150.0`

but this make crash firefox

**Comment by elulcao** [comment]
> > > I've added the code but infortunally it doesn't work, the same error appear.
> 
> > > I've see an error that appear also before adding the Interact=True
> 
> > > `/home/ubu/.local/lib/python3.9/site-packages/selenium/webdriver/remote/webelement.py:426: UserWarning: find_elements_by_* commands are deprecated. Please use find_elements() instead warnings.warn("find_elements_by_* commands are deprecated. Please use find_elements() instead")`
> 
> > 
> 
> > Working on it, check #6393
> 
> 
> 
> i have tested to change requirements.txt 
> 
> `Update requirements.txt from selenium>=3.141.0 to selenium>=3.141.0,<= 3.150.0`
> 
> 
> 
> but this make crash firefox
> 
> 
> 
> 

Did you installed the version 4 and reinstalled version 3? Right

**Comment by r4spberry13** [comment]
I just change the value in the requirments.txt

Selenenium was installed when i have launched pip install instapy

**Comment by r4spberry13** [comment]
I am have selenium 4.10 installed

**Comment by elulcao** [comment]
> I have selenium 4.10 installed

Downgrade to 3.150

**Comment by r4spberry13** [comment]
I have found 3.141 but impossible to install 3.150.0 also 3.150.1 and 3.150.2

```
pip install selenium==3.150.0
ERROR: Could not find a version that satisfies the requirement selenium==3.150.0
ERROR: No matching distribution found for selenium==3.150.0
```
maybe there's other alternative ?

**Comment by r4spberry13** [comment]
Is it possible the problem come from the language set  ?

My account is French and the language set for the web browser is English

**Comment by XBesi** [comment]
> 

This is happening at me. Im running Windows 11 And the latest update of firefox. Ive tried everything in this post but nothing works

**Comment by hetasanghvi** [comment]
This is happening to me.  I have tried all the available solutions none of those worked. Also, I have two Instagram business accounts. Everything works fine for one account but not for another.
